src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/OverloadResolution.cs

```diff
@@ -283,10 +284,15 @@ private static bool OverloadResolutionResultIsValid<TMember>(ArrayBuilder<Member
 
                 if (isMethodGroupConversion)
                 {
-                    RemoveDelegateConversionsWithWrongReturnType(results, ref useSiteDiagnostics, returnRefKind, returnType);
+                    RemoveDelegateConversionsWithWrongReturnType(results, ref useSiteDiagnostics, returnRefKind, returnType, isFunctionPointerResolution);
                 }
             }
 
+            if (isFunctionPointerResolution)
+            {
+                RemoveCallingConventionMismatches(results, callingConvention);
```

Comment:
>RemoveCallingConventionMismatches(results, callingConvention); [](start = 16, length = 62)

I am not sure if performing this filtering here is a good idea. The reason is that calling convention of methods is not part of the language, they are not in source, etc. If everything is filtered out, the user would get "there is no method" error, and how to figure out what is wrong. It might be better to pick the best candidate with no respect to the calling convention and then complain that it doesn't match due to that specific reason.

Reply:
> I am not sure if performing this filtering here is a good idea.

I'm not a huge fan of the naming conventions used in all of these `RemoveX` methods, as most of them aren't actually "removing" methods, they're just updating the result analysis to say that the method is not applicable for whatever reason. I kept the same style as the rest of the helpers, but I could change the name of the function if you prefer.

Reply:
Are we filtering out non-static methods early as well?

---
In reply to: [394032288](https://github.com/dotnet/roslyn/pull/42488#discussion_r394032288) [](ancestors = 394032288)

Reply:
> Are we filtering out non-static methods early as well?

No, but we probably should be at this stage. I'll add it here, and update the filtering method you mentioned below to "prefer" wrong calling conventions over static instance mismatches.

Reply:
We'll actually have to update the draft spec as well to do this: currently, we say that for a method `M` to be applicable, only the arguments must match. This means that, by today's draft spec, this should be ambiguous:

```cs
interface I1{}
interface I2{}

public unsafe class C : I1, I2 {
    void M(I1 i) {}
    static void M(I2 i) {}
    public void M1() {
        delegate*<C, void> a = M; // Ambiguous because both M's are applicable
    }
}
```

However, we know that `M(I1)` will never be compatible, so this should be resolvable. I'll update the spec to specify this as well.

Reply:
Submitted https://github.com/dotnet/csharplang/pull/3281 for this change.